provider/datagen/src/transform/cldr/decimal/compact_decimal_pattern.rs

```diff
@@ -307,9 +307,11 @@ impl TryFrom<&DecimalFormat> for CompactDecimalPatternDataV1<'static> {
                             index: pattern
                                 .placeholder
                                 .map_or(Some(u8::MAX), |p| {
-                                    u8::try_from(p.index)
-                                        .ok()
-                                        .and_then(|i| (i < u8::MAX).then(|| i))
+                                    u8::try_from(p.index).ok().and_then(|i| {
+                                        // Until we bump our MSVR above Rust 1.60
```

Comment:
```suggestion
                                        // Until we bump our MSRV above Rust 1.60
```

Also, we should add our MSRV to clippy.toml so clippy will just not complain about this

Reply:
Yes please (about the MSRV config)